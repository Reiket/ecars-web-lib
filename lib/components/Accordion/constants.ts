export const ACCORDION_TEST_ID = 'accordionTestId';
