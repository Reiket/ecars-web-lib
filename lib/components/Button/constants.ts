import type {ButtonHTMLAttributes} from 'react';

export const BUTTON_SIZE = Object.freeze({
  BIG: 'big',
  SMALL: 'small',
});

export const BUTTON_COLOR = Object.freeze({
  GREEN: 'green',
  WHITE: 'white',
  GRAY: 'gray',
});


export type ButtonAttributesType = ButtonHTMLAttributes<HTMLButtonElement>['type']
export type ButtonSizeType = (typeof BUTTON_SIZE)[keyof typeof BUTTON_SIZE];
export type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];

export const BUTTON_TEST_ID = 'buttonTestId';