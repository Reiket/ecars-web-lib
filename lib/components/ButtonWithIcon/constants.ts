export const LEFT_ICON_BUTTON_TEST_ID = 'leftIconButtonTestId';
export const RIGHT_ICON_BUTTON_TEST_ID = 'rightIconButtonTestId';
