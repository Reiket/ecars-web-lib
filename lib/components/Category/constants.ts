export const CATEGORIES_LIST = Object.freeze({
  GUIDES: 'guides',
  REVIEWS: 'reviews',
  NEWS: 'news',
});

export type CategoriesListType = (typeof CATEGORIES_LIST)[keyof typeof CATEGORIES_LIST];

export const CATEGORY_TEST_ID = 'categoryTestId';
