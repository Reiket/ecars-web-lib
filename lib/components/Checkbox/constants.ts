import type {ElementProps} from '@/services/types';

export interface CheckboxProps extends ElementProps {
  name?: string;
  id?: string;
  disabled?: boolean;
  onChange?: (checked: boolean) => void;
  checked?: boolean;
  hasError?: boolean;
}

export const CHECKBOX_TEST_ID = 'checkboxTestId';
export const CHECKBOX_TEST_INPUT_ID = 'checkboxTestInputId';
