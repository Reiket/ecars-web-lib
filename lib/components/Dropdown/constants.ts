import type {ReactNode} from 'react';

export interface DropdownOption {
  value: string;
  label?: ReactNode;
}

export interface DropdownProps {
  onSelect: (value: string) => void;
  category: string;
  options: DropdownOption[];
}

export const DROPDOWN_TEST_ID = 'dropdownTestId';

export const DROPDOWN_THEME = Object.freeze({
  DARK: 'dark',
  LIGHT: 'light',
});

export type DropdownThemeType = (typeof DROPDOWN_THEME)[keyof typeof DROPDOWN_THEME];
