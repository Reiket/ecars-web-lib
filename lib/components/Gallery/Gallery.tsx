import type {FC} from 'react';
import type {ReactImageGalleryItem} from 'react-image-gallery';
import ImageGallery from 'react-image-gallery';
import type {ElementProps} from '@/services/types';

interface Props extends ElementProps {
  images: ReactImageGalleryItem[];
}

export const Gallery: FC<Props> = ({images, block}) => {
  return (
    <ImageGallery
      items={images}
      additionalClass={block}
      showFullscreenButton={false}
      showPlayButton={false}
    />
  );
};
