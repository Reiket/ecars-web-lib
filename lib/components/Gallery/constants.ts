export const GALLERY_TEST_ID = 'galleryTestId';
