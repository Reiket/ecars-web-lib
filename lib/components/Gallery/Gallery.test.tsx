import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import {Gallery} from './Gallery';
import type {ReactImageGalleryItem} from 'react-image-gallery';
import {imagesMock} from '@/services/mocks';
import {GALLERY_TEST_ID} from '@/components/Gallery/constants';

jest.mock('react-image-gallery', () => {
  return ({items}: {items: ReactImageGalleryItem[]}) => (
    <div data-testid={GALLERY_TEST_ID}>
      {items.map((item, index) => (
        <img
          key={index}
          src={item.original}
          alt="image-gallery"
        />
      ))}
    </div>
  );
});

describe('Gallery component', () => {
  test('renders Gallery component', () => {
    const {container} = render(<Gallery images={imagesMock} />);
    const galleryElement = screen.getByTestId(GALLERY_TEST_ID);
    expect(galleryElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('renders all mock images', () => {
    render(<Gallery images={imagesMock} />);
    const images = screen.getAllByRole('img');
    expect(images).toHaveLength(imagesMock.length);
  });
});
