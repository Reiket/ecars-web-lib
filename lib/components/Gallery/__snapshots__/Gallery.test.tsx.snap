// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component renders Gallery component 1`] = `
<div>
  <div
    data-testid="galleryTestId"
  >
    <img
      alt="image-gallery"
      src="src/assets/react.svg"
    />
    <img
      alt="image-gallery"
      src="src/assets/react.svg"
    />
    <img
      alt="image-gallery"
      src="src/assets/react.svg"
    />
    <img
      alt="image-gallery"
      src="src/assets/react.svg"
    />
    <img
      alt="image-gallery"
      src="src/assets/react.svg"
    />
    <img
      alt="image-gallery"
      src="src/assets/react.svg"
    />
  </div>
</div>
`;
